astropy/io/fits/tests/test_hdulist.py

```diff
@@ -913,3 +914,38 @@ def test_proper_error_raised_on_non_fits_file_with_unicode(self):
         # This should raise an IOError because there is no end card.
         with pytest.raises(IOError):
             fits.open(filename)
+
+    # Avoid a spurious pyflakes warning about an unused import
+    # of six by adding a reason string
+    @pytest.mark.skipif(six.PY2, reason='six.PY2')
+    @pytest.mark.parametrize('ignore', [False, True])
+    @pytest.mark.parametrize('fileobj', [False, True])
```

Comment:
I actually don't understand this part. I skimmed through the conversation but none of it explains to me why `fileobj` needs to be a test input. Shouldn't you be checking to see if `filename` is a file object instead of an external input like this? Something like `if not isinstance(filename, six.string_types): ...  # a file object`. Of course, then you have to first create the file object at the start of the test. Am I missing something here?